test/topology_tasks/test_tablet_tasks.py

```diff
@@ -0,0 +1,130 @@
+    assert task.state in ["created", "running"]
+
+    status = await tm.get_task_status(servers[0].ip_addr, task.task_id)
+
+    check_repair_task_status(status)
+    assert status.state in ["created", "running"]
+
+    async def wait_for_task():
+        status_wait = await tm.wait_for_task(servers[0].ip_addr, task.task_id)
+        check_repair_task_status(status_wait)
+        assert status_wait.state == "done"
+
+    async def abort_task():
+        await tm.abort_task(servers[0].ip_addr, task.task_id)
+
+    await asyncio.gather(wait_for_task(), abort_task())
+
+@pytest.mark.asyncio
+@skip_mode('release', 'error injections are not supported in release mode')
+async def test_tablet_repair_task(manager: ManagerClient):
+    module_name = "tablets"
+    tm = TaskManagerClient(manager.api)
+
+    servers, cql, hosts, table_id = await create_table_insert_data_for_repair(manager)
+    assert module_name in await tm.list_modules(servers[0].ip_addr), "tablets module wasn't registered"
+
+    async def repair_task():
+        token = -1
+        # Keep retring tablet repair.
+        await inject_error_on(manager, "repair_tablet_fail_on_rpc_call", servers)
+        await manager.api.tablet_repair(servers[0].ip_addr, "test", "test", token)
+
+    await asyncio.gather(repair_task(), check_and_abort_repair_task(tm, servers, module_name))
+
+async def check_repair_task_list(tm: TaskManagerClient, servers: list[ServerInfo], module_name: str):
+    def get_task_with_id(repair_tasks, task_id):
+        tasks_with_id1 = [task for task in repair_tasks if task.task_id == task_id]
+        assert len(tasks_with_id1) == 1
+        return tasks_with_id1[0]
+
+    # Wait until user repair tasks are created.
+    repair_tasks0 = await wait_for_tasks(tm, servers[0], module_name, len(servers))
+    repair_tasks1 = await wait_for_tasks(tm, servers[1], module_name, len(servers))
+    repair_tasks2 = await wait_for_tasks(tm, servers[2], module_name, len(servers))
+
+    assert len(repair_tasks0) == len(repair_tasks1), f"Different number of repair virtual tasks on nodes {servers[0].server_id} and {servers[1].server_id}"
+    assert len(repair_tasks0) == len(repair_tasks2), f"Different number of repair virtual tasks on nodes {servers[0].server_id} and {servers[2].server_id}"
+
+    for task0 in repair_tasks0:
+        task1 = get_task_with_id(repair_tasks1, task0.task_id)
+        task2 = get_task_with_id(repair_tasks2, task0.task_id)
+
+        assert task0.table in ["test", "test2", "test3"]
+        assert task0.table == task1.table, f"Inconsistent table for task {task0.task_id}"
+        assert task0.table == task2.table, f"Inconsistent table for task {task0.task_id}"
+
+        for task in [task0, task1, task2]:
+            assert task.state in ["created", "running"]
+            assert task.type == "user_repair"
+            assert task.kind == "cluster"
+            assert task.scope == "table"
+            assert task.keyspace == "test"
```

Comment:
nit: Perhaps we can reuse check_repair_task_status above, e.g., 

check_repair_task_status(expected_states(['created', 'running']))

Reply:
Those are different types (TaskStatus vs TaskStats) and I would rather not mix them

Reply:
ok